doc/architecture.md

```diff
@@ -0,0 +1,38 @@
+# k8s-claimer architecture
+
+This document details how the k8s claimer server performs its lease and release operations in
+accordance with the guarantees it gives.
+
+## General
+
+Each cluster is assigned a unique name, and the GKE API/backend stores those names (along
```

Comment:
Nope, each lease is assigned a uuid (referred to as a token in the design doc). The cluster is assigned a static (but unique) name, which doesn't change between leases.